### DISCUSSION TITLE
How to define the title of mdx in jsx components when using mdx, so that toc can recognize and generate it

### DISCUSSION THREAD

**Discussion by hzihhuang-site** [web-infra-dev/rspress#1560] [Q&A]
<img width="891" alt="image" src="https://github.com/user-attachments/assets/cdaf8401-7773-4bee-8fcc-53f65e7d11b4">

**Comment by Timeless0911** [comment]
The collection of TOC cannot complete in components currently, it only collects title in mdast.

https://github.com/web-infra-dev/mdx-rs/blob/main/crates/plugin_toc/src/lib.rs

**Reply by hzihhuang-site** [reply]
Will it be supported in the future?
Sometimes it may be more convenient to use it in a component.

<img width="407" alt="image" src="https://github.com/user-attachments/assets/5700e242-95aa-4b3e-938d-80732ae170ab">

**Reply by Timeless0911** [reply]
This is a challenging feature which is a little bit hard to implement (deal `Node::MdxJsxElement` in mdast) and we have no plans in the short term. If you are interested, PR is welcome for now.
